src/rt_core_v2/rttuple_formatter.py

```diff
@@ -1,13 +1,22 @@
-import json, enum
+import json
+import enum
+from io import StringIO
+
+from rt_core_v2.rttuple import TupleComponents, TupleType, type_to_class
 
 def rttuple_to_json(input_rttuples):
     """Converts either a list or an RtTuple to a json object"""
-    return json.dumps(input_rttuples.get_str_attributes())
+    return json.dumps(input_rttuples.get_str_attributes(), default=lambda o: o.toJSON() if hasattr(o, 'toJSON') else str(o))
```

Comment:
> json.dumps(rttuple.**__dict__**)
This will not work, as Rui is not json serializable. The options are to either have a custom function in each class to let them to decide how they will be converted to a json or (the better option) to use a custom json encoder. 

I will swap to a custom encoder to remove the dependency in a future version.